### PR TITLE
added test for sendCalls and getCallsStatus

### PR DESCRIPTION
**Pull request by KannuSingh**
Added test component in laboratory for ethers and wagmi for EIP5792 methods sendCalls and getCallsStatus

### CODE REVIEW

**Comment by glitch-txs** [comment]
let's wait for Rocky approval before merging 👍

File: apps/laboratory/src/components/Ethers/EthersGetCallsStatusTest.tsx
Comment (tomiir) [inline apps/laboratory/src/components/Ethers/EthersGetCallsStatusTest.tsx:16]: isn't this setting a boolean flag on `ethereumProvider` ? Or is it resolving to the typed version of `walletProvider`?

If that's the case I believe it'd be best to do the check + cast whenever this `ethereumProvider` is used directly.

File: apps/laboratory/src/components/Ethers/EthersGetCallsStatusTest.tsx
Comment (tomiir) [inline apps/laboratory/src/components/Ethers/EthersGetCallsStatusTest.tsx:70]:
```suggestion
        Wallet does not support this feature
```

File: apps/laboratory/src/components/Ethers/EthersGetCallsStatusTest.tsx
Comment (tomiir) [inline apps/laboratory/src/components/Ethers/EthersGetCallsStatusTest.tsx:80]: same

File: apps/laboratory/src/components/Ethers/EthersSendCallsTest.tsx
Comment (tomiir) [inline apps/laboratory/src/components/Ethers/EthersSendCallsTest.tsx:15]: same

File: apps/laboratory/src/components/Ethers/EthersSendCallsTest.tsx
Comment (tomiir) [inline apps/laboratory/src/components/Ethers/EthersSendCallsTest.tsx:85]: does

File: apps/laboratory/src/components/Ethers/EthersSendCallsTest.tsx
Comment (tomiir) [inline apps/laboratory/src/components/Ethers/EthersSendCallsTest.tsx:95]: same

File: apps/laboratory/src/components/Wagmi/WagmiSendCallsTest.tsx
Comment (tomiir) [inline apps/laboratory/src/components/Wagmi/WagmiSendCallsTest.tsx:71]: I'd move this outside the effect

File: apps/laboratory/src/components/Wagmi/WagmiGetCallsStatusTest.tsx
Comment (tomiir) [inline apps/laboratory/src/components/Wagmi/WagmiGetCallsStatusTest.tsx:55]: I'd move this outside the effect

File: apps/laboratory/src/types/EIP5792.ts
Comment (tomiir) [inline apps/laboratory/src/types/EIP5792.ts:7]: what would be the key to this record? an address?

File: packages/wagmi/src/utils/defaultWagmiReactConfig.ts
Comment (tomiir) [inline packages/wagmi/src/utils/defaultWagmiReactConfig.ts:5]: is this change needed?

**Review by tomiir — COMMENTED** [review]
Looks awesome! Minor comments related to some strings and general code cleanup but overall looks 💯

File: apps/laboratory/src/types/EIP5792.ts
Comment (KannuSingh) [reply apps/laboratory/src/types/EIP5792.ts:7]: chainId in hex

File: packages/wagmi/src/utils/defaultWagmiReactConfig.ts
Comment (KannuSingh) [reply packages/wagmi/src/utils/defaultWagmiReactConfig.ts:5]: I guess, I did this bc of I was getting the wrong dependency when not updated the wagmi version across the project but only for this package and not sure why the import are different for `defaultWagmiCoreConfig.ts` and `defaultWagmiReactConfig.ts`. 
Let me know if this is not needed.

File: apps/laboratory/package.json
Comment (tomiir) [inline apps/laboratory/package.json:51]: connectors is to force the proper version of the provider, correct?
but why remove core?

**Review by tomiir — APPROVED** [review]

File: apps/demo/tsconfig.json
Comment (enesozturk) [inline apps/demo/tsconfig.json:25]: Why did we add this? maybe something was automatically added?

File: apps/laboratory/src/components/Ethers/EthersSendCallsTest.tsx
Comment (enesozturk) [inline apps/laboratory/src/components/Ethers/EthersSendCallsTest.tsx:105]: Could we avoid inline array calls like this? 

```
const allowedChainNames = allowedChains.map(ci => ci.chainName).join(', ');
const currentChainInfo = allowedChains.find(chainInfo => chainInfo.chainId === Number(chainId));

...

return currentChainInfo ? (
  <Stack direction={['column', 'column', 'row']}>
    ...
  </Stack>
) : (
  <Text fontSize="md" color="yellow">
    Switch to {allowedChainNames} to test this feature
  </Text>
)
```

File: apps/laboratory/src/components/Ethers/EthersSendCallsTest.tsx
Comment (enesozturk) [inline apps/laboratory/src/components/Ethers/EthersSendCallsTest.tsx:16]: Separate parts of the component with new line. Basically `state hooks/other hooks/functions/render` is good approach to separate them.

```suggestion

  async function onSendCalls() {
```

File: apps/laboratory/src/components/Wagmi/WagmiSendCallsTest.tsx
Comment (enesozturk) [inline apps/laboratory/src/components/Wagmi/WagmiSendCallsTest.tsx:132]: Same here

File: apps/laboratory/src/utils/CommonUtils.ts
Comment (enesozturk) [inline apps/laboratory/src/utils/CommonUtils.ts:9]: Seems like `value` already could be `bigint`, why do we need conditional return in the body? Is it unknown or non-bigint in the place where this function used?

File: apps/demo/tsconfig.json
Comment (KannuSingh) [reply apps/demo/tsconfig.json:25]: It got automatically added. Should I remove it.

File: apps/laboratory/src/utils/CommonUtils.ts
Comment (KannuSingh) [reply apps/laboratory/src/utils/CommonUtils.ts:9]: It need to be change to any, and if it type is bigint, then value.toString. I will do the change.

File: apps/laboratory/package.json
Comment (KannuSingh) [reply apps/laboratory/package.json]: added the core dependency as it is being used.

File: apps/demo/tsconfig.json
Comment (enesozturk) [reply apps/demo/tsconfig.json:25]: I think yeah, bundled code shouldn't be type checked.

File: apps/laboratory/src/utils/CommonUtils.ts
Comment (enesozturk) [reply apps/laboratory/src/utils/CommonUtils.ts]: @KannuSingh I'm not sure if it should be `any`, don't we know possible types of the parameter we pass to this function? If it's not known, we could use `unknown`, otherwise we should keep as strict as possible.

Comment (KannuSingh) [reply apps/laboratory/src/utils/CommonUtils.ts]: it is used in JSON.stringify(batchCallsStatus, bigIntReplacer).


```
(method) JSON.stringify(value: any, replacer?: ((this: any, key: string, value: any) => any) | undefined, space?: string | number | undefined): string 
```

File: apps/demo/tsconfig.json
Comment (KannuSingh) [reply apps/demo/tsconfig.json]: Done

File: apps/laboratory/src/components/Ethers/EthersSendCallsTest.tsx
Comment (KannuSingh) [reply apps/laboratory/src/components/Ethers/EthersSendCallsTest.tsx]: Done

File: apps/laboratory/src/components/Ethers/EthersSendCallsTest.tsx
Comment (KannuSingh) [reply apps/laboratory/src/components/Ethers/EthersSendCallsTest.tsx:29]: Can you check if this done now?

File: apps/laboratory/src/components/Wagmi/WagmiSendCallsTest.tsx
Comment (KannuSingh) [reply apps/laboratory/src/components/Wagmi/WagmiSendCallsTest.tsx]: Done

**Review by enesozturk — APPROVED** [review]

File: apps/laboratory/src/components/Ethers/EthersGetCallsStatusTest.tsx
Comment (KannuSingh) [reply apps/laboratory/src/components/Ethers/EthersGetCallsStatusTest.tsx]: Done

File: apps/laboratory/src/components/Ethers/EthersGetCallsStatusTest.tsx
Comment (KannuSingh) [reply apps/laboratory/src/components/Ethers/EthersGetCallsStatusTest.tsx]: Done

File: apps/laboratory/src/components/Ethers/EthersSendCallsTest.tsx
Comment (KannuSingh) [reply apps/laboratory/src/components/Ethers/EthersSendCallsTest.tsx]: Done

**Pull request merged**
